### PR TITLE
feat(enhancement): Support outfit (un)installation costs

### PR DESCRIPTION
**Pull request by TomGoodIdea**
**Feature:** This PR implements the feature request detailed in issue #6623

## Feature Details
Outfits can set `"installation cost"` and `"uninstallation cost"` attributes. The costs won't affect buying directly into cargo, moving items from cargo to planetary storage, plundering etc., only installing these oufits on ships at the outfitter.

## UI Screenshots
(The pictures show example values that are not included in the changes.)
An (un)installation cost set to 0 (default) doesn't get listed at all.

ex1

ex2

## Testing Done
When I had needed amount of credits, the process succeded and the credits were consumed. When I didn't have the credits, the process was blocked and a dialog box was displayed.

## Performance Impact
N/A

### CODE REVIEW

File: source/OutfitterPanel.cpp [hunk 1]
```diff
@@ -424,6 +424,14 @@ ShopPanel::BuyResult OutfitterPanel::CanBuy(bool onlyOwned) const
...
 	else
 	{
+		// Installing on a ship, so check if the player can pay for installation.
+		int64_t credits = player.Accounts().Credits();
+		int64_t installationCost = selectedOutfit->InstallationCost();
+		if(installationCost > credits)
+			return "You cannot install this outfit, because the installation process costs "
+				+ Format::CreditString(installationCost) + ", and you only have "
+				+ Format::Credits(credits) + ".";
+
```
Comment (mOctave) [inline source/OutfitterPanel.cpp:434]: This should help with stylecheck.

```suggestion

```

**Comment by quyykk** [comment]
I can see why this feature would be nice to have, but I have a concern because it's kinda of a hidden cost, and it can drain a player's credits if they're experimenting with loadouts if that makes sense.

**Comment by mOctave** [comment]
Maybe it would be nice to have a confirmation dialog when you're installing saying "these things will cost money to uninstall" and when you're uninstalling saying "you'll need to pay X amount to uninstall these"

**Comment by kestrel1110** [comment]
A better way to do this would be to have the costs, but instead allow a player to install/uninstall outfits as much as they wanted
Then upon exiting the outfitter, a confirmation dialogue would show to inform the player of the costs deducted

That way would keep the level of experimentation ES allows while also adding this feature that allows for more diversity of outfits

**Comment by bene-dictator** [comment]
Does this cost occurs each time you install or uninstall an outfit? So you could buy that Sidewinder Missile Launcher, install it (+1k), uninstall it to cargo (+1k), install it again (+1k) for a cost of 3k? Or is it a one-off cost (meaning the example would be only 2k credits)?

In addition, can you modify these costs with events and licences? I'm not sure if you can modify the cost of an existing outfit, too, but it might be useful.

> A better way to do this would be to have the costs, but instead allow a player to install/uninstall outfits as much as they wanted Then upon exiting the outfitter, a confirmation dialogue would show to inform the player of the costs deducted
> 
> That way would keep the level of experimentation ES allows while also adding this feature that allows for more diversity of outfits

I don't see how this would solve anything because it would lead people to question why they now have a huge bill they might have not realised was being given to them until the dialogue box at the end. Which is why I want to clarify.

In any case, save-scumming would probably mean most people would save whatever wasted credits.

**Comment by mOctave** [comment]
> >A better way to do this would be to have the costs, but instead allow a player to install/uninstall outfits as much as they wanted Then upon exiting the outfitter, a confirmation dialogue would show to inform the player of the costs deducted
>>
> >That way would keep the level of experimentation ES allows while also adding this feature that allows for more diversity of outfits
>
>I don't see how this would solve anything because it would lead people to question why they now have a huge bill they might have not realised was being given to them until the dialogue box at the end. Which is why I want to clarify.

I think the plan with that would be to only apply the installation and uninstallation costs when leaving the outfitter. It might be hard to set things up that way, but it would definitely be a nice approach if it worked.

**Comment by Amazinite** [comment]
Not a fan of the idea behind these attributes. As noted, the current implementation of applying the cost every time you install or uninstall the outfit is bad for experimentation, but at the same time I feel like the suggested alternative of having a confirmation dialog upon leaving the outfitter to apply these costs is overengineering it. And gameplay wise, what would we even gain by having such attributes? It seems very odd to me to be able to have what is effectively a punishment for installing or uninstalling an outfit.

**Pull request closed without merging**